### PR TITLE
Add package readme for Microsoft.Extensions.Configuration.Binder

### PR DESCRIPTION
**Pull request by MSDN-WhiteKnight**
Issue: #59630

### CODE REVIEW

**Comment by tarekgh** [comment]
@MSDN-WhiteKnight thanks for submitting that. Could you please follow the same format for the readmes. Here is exampel https://github.com/dotnet/runtime/tree/main/src/libraries/Microsoft.Extensions.Caching.Abstractions#readme.

File: src/libraries/Microsoft.Extensions.Configuration.Binder/src/README.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+Documentation can be found at https://learn.microsoft.com/dotnet/core/extensions/configuration
+
+## Contribution Bar
+- [x] [We consider new features, new APIs, bug fixes, and performance changes](../../../libraries/README.md#primary-bar)
+
+The APIs and functionality are mature, but do get extended occasionally.
+
+## Deployment
+[Microsoft.Extensions.Configuration.Binder](https://www.nuget.org/packages/Microsoft.Extensions.Configuration.Binder/) is not included in the shared framework. The package is deployed as out-of-band (OOB) and needs to be installed into projects directly.
```
Comment (tarekgh) [inline src/libraries/Microsoft.Extensions.Configuration.Binder/src/README.md:13]: >  is not included in the shared framework

Could you please change this to be ` is not included in the ASP.NET core shared framework`.

We'll need to fix this in the rest of readmes.

**Comment by tarekgh** [comment]
Could you please move the readme to the root of the library folder?

File: src/libraries/Microsoft.Extensions.Configuration.Binder/README.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+Documentation can be found at https://learn.microsoft.com/dotnet/core/extensions/configuration
+
+## Contribution Bar
+- [x] [We consider new features, new APIs, bug fixes, and performance changes](../../libraries/README.md#primary-bar)
+
+The APIs and functionality are mature, but do get extended occasionally.
+
+## Deployment
+[Microsoft.Extensions.Configuration.Binder](https://www.nuget.org/packages/Microsoft.Extensions.Configuration.Binder/) is not included in the shared framework. The package is deployed as out-of-band (OOB) and needs to be installed into projects directly.
```
Comment (tarekgh) [inline src/libraries/Microsoft.Extensions.Configuration.Binder/README.md:13]: [Microsoft.Extensions.Configuration.Binder](https://www.nuget.org/packages/Microsoft.Extensions.Configuration.Binder/) is included in the ASP.NET core shared framework. The package is deployed as out-of-band (OOB) too and can be referenced into projects directly.

File: src/libraries/Microsoft.Extensions.Configuration.Binder/src/README.md (hunk 1 above)
Comment (MSDN-WhiteKnight) [reply src/libraries/Microsoft.Extensions.Configuration.Binder/src/README.md]: I assume you meant "is included..."? It's actually there looking at shipped binaries. But it is not included in regular Netcoreapp shared framework. So shouldn't it be something like this:

> [Microsoft.Extensions.Configuration.Binder](https://www.nuget.org/packages/Microsoft.Extensions.Configuration.Binder/) is included in ASP.NET Core shared framework, but not in regular .NET shared framework. The package is deployed as out-of-band (OOB) and needs to be installed into projects directly when not using ASP.NET Core.

**Comment by tarekgh** [comment]
I left one last comment. LGTM otherwise.

File: src/libraries/Microsoft.Extensions.Configuration.Binder/src/README.md (hunk 1 above)
Comment (MSDN-WhiteKnight) [reply src/libraries/Microsoft.Extensions.Configuration.Binder/src/README.md]: Updated, didn't noticed the latest review suggestion when commenting.

Comment (tarekgh) [reply src/libraries/Microsoft.Extensions.Configuration.Binder/src/README.md]: I confirmed this library is part of ASP.NET Core Runtime

**Review by tarekgh — APPROVED** [review]

**Comment by tarekgh** [comment]
I'll wait CI to finish then will merge this one.

**Comment by MSDN-WhiteKnight** [comment]
>Are you interested in providing more readme files for Microsoft.Extensions libraries? it is easy to check which library is missing the readme.

Yes, i'll pick `Microsoft.Extensions.Configuration.*` packages without readme then. I've also asked a question about the intended direction with readmes there: <https://github.com/dotnet/runtime/pull/77413#issuecomment-1296324839> (not sure if commenting under merged PR is good idea, but i don't see the issue that discusses this).

**Pull request merged**